backend/amd64/emit.mlp

```diff
@@ -587,47 +596,87 @@ let emit_float_test cmp i ~(taken:X86_ast.condition -> unit) =
      If FP traps are on (they are off by default),
      comisd traps on QNaN and SNaN but ucomisd traps on SNaN only.
   *)
-  match cmp with
-  | CFeq when arg i 1 = arg i 0 ->
+  match width, cmp with
+  | Float64, CFeq when arg i 1 = arg i 0 ->
       I.ucomisd (arg i 1) (arg i 0);
       taken NP
-  | CFeq ->
+  | Float64, CFeq ->
       let next = new_label() in
       I.ucomisd (arg i 1) (arg i 0);
       I.jp (label next);           (* skip if unordered *)
       taken E;                     (* branch taken if x=y *)
       def_label next
-  | CFneq when arg i 1 = arg i 0 ->
+  | Float64, CFneq when arg i 1 = arg i 0 ->
       I.ucomisd (arg i 1) (arg i 0);
       taken P
-  | CFneq ->
+  | Float64, CFneq ->
       I.ucomisd (arg i 1) (arg i 0);
       taken P;                     (* branch taken if unordered *)
       taken NE                     (* branch taken if x<y or x>y *)
-  | CFlt ->
+  | Float64, CFlt ->
       I.comisd (arg i 0) (arg i 1);
       taken A                      (* branch taken if y>x i.e. x<y *)
-  | CFnlt ->
+  | Float64, CFnlt ->
       I.comisd (arg i 0) (arg i 1);
       taken BE                     (* taken if unordered or y<=x i.e. !(x<y) *)
-  | CFle ->
+  | Float64, CFle ->
       I.comisd (arg i 0) (arg i 1);(* swap compare *)
       taken AE                     (* branch taken if y>=x i.e. x<=y *)
-  | CFnle ->
+  | Float64, CFnle ->
       I.comisd (arg i 0) (arg i 1);(* swap compare *)
       taken B                      (* taken if unordered or y<x i.e. !(x<=y) *)
-  | CFgt ->
+  | Float64, CFgt ->
       I.comisd (arg i 1) (arg i 0);
       taken A                      (* branch taken if x>y *)
-  | CFngt ->
+  | Float64, CFngt ->
       I.comisd (arg i 1) (arg i 0);
       taken BE                     (* taken if unordered or x<=y i.e. !(x>y) *)
-  | CFge ->
+  | Float64, CFge ->
       I.comisd (arg i 1) (arg i 0);(* swap compare *)
       taken AE                     (* branch taken if x>=y *)
-  | CFnge ->
+  | Float64, CFnge ->
       I.comisd (arg i 1) (arg i 0);(* swap compare *)
       taken B                      (* taken if unordered or x<y i.e. !(x>=y) *)
+  | Float32, CFeq when arg i 1 = arg i 0 ->
+      I.ucomisd (arg i 1) (arg i 0);
+      taken NP
+  | Float32, CFeq ->
+      let next = new_label() in
+      I.ucomisd (arg i 1) (arg i 0);
+      I.jp (label next);           (* skip if unordered *)
+      taken E;                     (* branch taken if x=y *)
+      def_label next
+  | Float32, CFneq when arg i 1 = arg i 0 ->
```

Comment:
It looks like a copy/paste from the
`Float64` cases, only changing the
`I.ucomisd` call to `I.ucomiss`. If it
is the case, it might be worth
factoring out.